tests/std/tests/P1502R1_standard_library_header_units/test.cpp

```diff
@@ -0,0 +1,973 @@
+        } // destroying jt will ask it to stop
+        static constexpr int expected[]{1729, 5188, 2594, 1297, 3892, 1946, 973, 2920, 1460, 730, 365, 1096, 548, 274,
+            137, 412, 206, 103, 310, 155, 466, 233, 700, 350, 175, 526, 263, 790, 395, 1186, 593, 1780, 890, 445, 1336,
+            668, 334, 167, 502, 251, 754, 377, 1132, 566, 283, 850, 425, 1276, 638, 319, 958, 479, 1438, 719, 2158,
+            1079, 3238, 1619, 4858, 2429, 7288, 3644, 1822, 911, 2734, 1367, 4102, 2051, 6154, 3077, 9232, 4616, 2308,
+            1154, 577, 1732, 866, 433, 1300, 650, 325, 976, 488, 244, 122, 61, 184, 92, 46, 23, 70, 35, 106, 53, 160,
+            80, 40, 20, 10, 5, 16, 8, 4, 2, 1, -1000};
+        assert(equal(vec.begin(), vec.end(), begin(expected), end(expected)));
+    }
+
+    {
+        puts("Testing <streambuf>.");
+        istringstream iss{"kittens"};
+        assert(iss.rdbuf()->in_avail() == 7);
+    }
+
+    {
+        puts("Testing <string_view>.");
+        constexpr string_view catenary{"catenary"};
+        assert(catenary.starts_with("cat"));
+        assert(!catenary.starts_with("dog"));
+        static_assert(catenary.starts_with("cat"));
+        static_assert(!catenary.starts_with("dog"));
+    }
+
+    {
+        puts("Testing <string>.");
+        const string small_string{"homeowner"};
+        const string large_string{"Cute fluffy kittens are so adorable when they meow and purr."};
+        assert(small_string.find("meow") == 2);
+        assert(large_string.find("meow") == 46);
+    }
+
+    {
+        puts("Testing <strstream>.");
+        istrstream istr{"1729"};
+        int n = -1;
+        istr >> n;
+        assert(n == 1729);
+    }
+
+    {
+        puts("Testing <syncstream>.");
+        puts("(TRANSITION, not yet implemented.)");
+    }
+
+    {
+        puts("Testing <system_error>.");
+        const error_code code = make_error_code(errc::value_too_large);
+        assert(code.value() == static_cast<int>(errc::value_too_large));
+        assert(code.category() == generic_category());
+        assert(code);
+    }
+
+    {
+        puts("Testing <thread>.");
+        // see above, tested with <condition_variable>
+        assert(this_thread::get_id() != thread::id{});
+    }
+
+    {
+        puts("Testing <tuple>.");
+        constexpr tuple<int, char, double> t{1729, 'c', 1.25};
+        assert(get<int>(t) == 1729);
+        assert(get<char>(t) == 'c');
+        assert(get<double>(t) == 1.25);
+        static_assert(get<int>(t) == 1729);
+        static_assert(get<char>(t) == 'c');
+        static_assert(get<double>(t) == 1.25);
+    }
+
+    {
+        puts("Testing <type_traits>.");
+        static_assert(is_void_v<void>);
+        static_assert(!is_void_v<double>);
+        static_assert(is_same_v<remove_extent_t<int[10][20][30]>, int[20][30]>);
+        static_assert(is_same_v<remove_extent_t<double>, double>);
+
+        assert(compiletime_10_or_runtime_20() == 20);

```

Comment:
Should `compiletime_10_or_runtime_20` be a lambda so we can define it closer to the point of use?

Reply:
Yes, this works nicely. Thanks, done.